### ISSUE TITLE
Resize observer

### ISSUE THREAD

**Issue by mattnolting** [patternfly/patternfly#6660]
# TLDR

Container queries look like an ideal replacement for media-queries, but there are unforeseen implications in their use. IMO we need to

[https://github.com/RedHatInsights/landing-page-frontend/compare/master...Hyperkid123:landing-page-frontend:resize-POC](https://github.com/RedHatInsights/landing-page-frontend/compare/master...Hyperkid123:landing-page-frontend:resize-POC)

- Avoid applying extra DOM elements

Related docs: 

[https://docs.google.com/document/d/1uEPqBg2rDDZghfmuyFv3dgpuC1ALHdgYxyvWAzVFK7M/edit](https://docs.google.com/document/d/1uEPqBg2rDDZghfmuyFv3dgpuC1ALHdgYxyvWAzVFK7M/edit)

**Describe the feature**
Leverages resize observer and vanilla JS 

**Cons of container queries** note: these cons apply to the container itself being unaware of its own size

- Container queries require a superfluous wrapping element. Anything placed specifically in a page layout is now broken.

- In the context of an enterprise applications, this construct would add a tremendous amount of DOM bloat.

- Container queries are immutable and therefore can't be updated to suit varied use cases.

- ~~Can't use absolute positioning on child elements~~ 

- Specificity issues with adding container classes

- Browsers continuously monitor container queries, even more performance impact.

Reference: #5111

---

**Jira Issue:** [PF-2239](https://redhat.atlassian.net/browse/PF-2239)

**Comment by garrett** [comment]
I strongly disagree with this approach (adding more JavaScript instead of doing what browsers do natively), and have been waiting on the proper solution from browser vendors for several years and PatternFly for a while, as this is something we've needed in Cockpit for years. 

The least invasive and most web-native approach would be to use CSS size queries, which I outline @ https://github.com/patternfly/patternfly/issues/5111#issuecomment-2114277854

The quick summary would be:

1. Add a reference to the container, so the media queries have something to compare the size.
2. Change media queries to container queries with a regular expression.
3. Adjust the magic numbers (to subtract the sidebar).

That's it! It would then be a drop-in replacement to what PatternFly currently does, and it wouldn't need any JavaScript, and it'd work on sites and apps that have a sidebar as well as ones that don't.

It could (optionally) further be extended to optimize the queries (as we don't need them to be inclusive, but can be less than or greater than without equal), which would actually fix some race condition sizing bugs inherent to old-style media queries and simplify the code. And it could include other more localized media queries. But it doesn't need any of this; just switching existing media queries to container size queries would solve the problem.

Additionally: There was also no discussion with examples for why resize observers are being considered. There really should be some kind of justification instead of just closing a years-long discussion for something that's needed that browsers now provide a native solution for — unless it was a mistake and/or the proper native solution wasn't understood. If so, then we can continue the discussion for the native approach (container size queries) instead of just dead-ending it without any explanation or examples.

**Comment by garrett** [comment]
Still an issue that should be addressed.
